### ISSUE TITLE
compiler crash with  clang-10: /home/neboat/opencilk/llvm/include/llvm/ADT/ArrayRef.h:250: const T &llvm::ArrayRef<llvm::BasicBlock *>::operator[](size_t) const [T = llvm::BasicBlock *]: Assertion `Index < Length && "Invalid index!"' failed.

### ISSUE THREAD

**Issue by wheatman** [OpenCilk/opencilk-project#73]
The following code causes a crash, I would expect this minimal example to just give me some linker errors.  I was starting to use the std::array when I first triggered the crash so I would guess it is something with that. 

```

#include <array>
#include <cilk/cilk.h>

class inner {
public:
  inner();
};

class outer {

public:
  outer() = default;
  outer(const outer &source) {
    cilk_for(int i = 0; i < 100; i++) { new inner(); }
  }
  // crash
  // ~outer();
  ~outer(){};
  // don't crash
  // ~outer() = default;
};

int main() {

  outer g;
  std::array<outer, 2> gs = {g, g};

  return 0;
}

```

I am compiling it with `clang++  -O2  -std=c++17 -fopencilk   run.cpp -lopencilk` it also crashes with O3, but not O0, O1, or Os

It triggers the assertion 
```
clang-10: /home/neboat/opencilk/llvm/include/llvm/ADT/ArrayRef.h:250: const T &llvm::ArrayRef<llvm::BasicBlock *>::operator[](size_t) const [T = llvm::BasicBlock *]: Assertion `Index < Length && "Invalid index!"' failed.
```

The associated files are attached 
[run-10b9bd.sh.txt](https://github.com/OpenCilk/opencilk-project/files/7357944/run-10b9bd.sh.txt)
[run-10b9bd.cpp.txt](https://github.com/OpenCilk/opencilk-project/files/7357945/run-10b9bd.cpp.txt)
[log.txt](https://github.com/OpenCilk/opencilk-project/files/7357946/log.txt)

```
clang++  --version
clang version 10.0.1 (git@github.com:OpenCilk/opencilk-project.git 8435006eea5d32f9cf895a666496b89613e2cbc1)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/ubuntu/OpenCilk-10.0.1-Linux/bin
```

**Comment by neboat** [comment]
Thanks for the bug report.  I should have a fix for the next OpenCilk release.

**Comment by neboat** [comment]
This issue has been fixed in version 1.1.
